### PR TITLE
Code Style Improvements

### PR DESCRIPTION
**Pull request by chrisbenincasa**
_In Progress_
- Updates to new version of Crane which uses `ClassTag` to return downcasted types rather than `Entity`.
  - This reduces a lot of redundant code and type-checking that had to be done whenever a call to `getComponent` was made
- Reduce build warnings
- Make some thing more idiomatic
  - remove redundancies
    - declaring `val` in `case class` parameter (this is implicit)
  - `var` -> `val` when value is immutable
  - remove a bunch of `null` and replace with `Option` methods
  - replaces `for` loops with `foreach` when it aids readability
  - empty-parens when function has side-effects
- Style improvements galore
  - spacing for parenthesis
  - brackets when pattern matching

This is still in progress and I'll be pushing up more soon. The code is getting much more readable, in my opinion.

### CODE REVIEW

File: src/main/scala/ayai/actions/MoveDirection.scala [hunk 1]
```diff
@@ -41,23 +41,25 @@ case object RightDirection extends MoveDirection(1,0) {
   }
 }
 
-case object UpDirection extends MoveDirection(0,-1){
+case object UpDirection extends MoveDirection(0,-1) {
```
Comment (timothyhahn) [inline src/main/scala/ayai/actions/MoveDirection.scala:44]: [nit] space after comma

File: src/main/scala/ayai/systems/ai/mapgeneration/FractionalBrownianMotion.scala [hunk 2]
```diff
@@ -25,19 +26,20 @@ class FractionalBrownianMotionGenerator(val noiseGenerator: NoiseGenerator) {
     var amplitude = 0.95
 
     //The first two octaves are skipped, so subtract 2
-    var octaves = floor(log(min(width, height))/log(2)).toInt - 2
+    val octaves = floor(log(min(width, height))).toInt - 2
```
Comment (chrisbenincasa) [inline src/main/scala/ayai/systems/ai/mapgeneration/FractionalBrownianMotion.scala]: revert this change
